src/main/java/org/broadinstitute/hellbender/utils/Nucleotide.java

```diff
@@ -3,22 +3,49 @@
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
-import java.util.stream.LongStream;
 
 /**
  * Represents the nucleotide alphabet with support for IUPAC ambiguity codes.
  *
  * <p>
  *    This enumeration not only contains standard (non-ambiguous) nucleotides, but also
- *    values to represent ambiguous and invalid codes.
+ *    values to represent ambiguous and an invalid nucleotide call ({@link #X} aka {@link #INVALID}.
  * </p>
  *
+ * <p>
+ *     You can query whether a value refers to a non-ambiguous nucleotide with {@link #isStandard()} or
+ *     {@link #isAmbiguous()} depending of your preference. Notice that the special value {@link #X}
+ *     is neither of those.
+ * </p>
+ *
+ * <p>
+ *     Querying the {@link #X} value for its {@link #complement}, {@link #transition} or
+ *     {@link #transversion} or using it in other operations
+ *     likes {@link #intersect} would result in returning also a {@link #X}; similar to {@link Double#NaN} in
```

Comment:
likes intersect would result in returning also a X -> like (or perhaps "such as") intersect will return an X